src/com/xilinx/rapidwright/examples/CounterGenerator.java

```diff
@@ -0,0 +1,277 @@
+            accepts(CLK_CONSTRAINT_OPT).withOptionalArg().ofType(Double.class).defaultsTo(clkPeriodConstraint).describedAs("Clk period constraint (ns)");
+            accepts(WIDTH_OPT).withOptionalArg().ofType(Integer.class).defaultsTo(width).describedAs("Operand width");
+            accepts(SLICE_SITES_OPT).withOptionalArg().defaultsTo(sliceSite).describedAs("Lower left slice to be used for adder/subtracter");
+            accepts(COUNT_DOWN_OPT).withOptionalArg().ofType(Boolean.class).defaultsTo(countDown).describedAs("Counts down instead of up");
+            accepts(STEP_OPT).withOptionalArg().ofType(Long.class).defaultsTo(step).describedAs("Counts down instead of up");
+            accepts(INIT_OPT).withOptionalArg().ofType(Long.class).defaultsTo(initValue).describedAs("Counts down instead of up");
+            accepts(VERBOSE_OPT).withOptionalArg().ofType(Boolean.class).defaultsTo(verbose).describedAs("Print verbose output");
+            acceptsAll( Arrays.asList(HELP_OPT, "?"), "Print Help" ).forHelp();
+        }};
+
+        return p;
+    }
+
+    private static void printHelp(OptionParser p) {
+        MessageGenerator.printHeader("Counter Generator");
+        System.out.println("This RapidWright program creates a placed and routed DCP for UltraScale+ devices that contains \n"
+                + "imported into UltraScale+ designs to perform addition or subtraction.  See \n"
+                + "RapidWright documentation for more information.\n");
+        try {
+            p.accepts(OUT_DCP_OPT).withOptionalArg().defaultsTo("counter.dcp").describedAs("Output DCP File Name");
+            p.printHelpOn(System.out);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Implements a parameterizable counter for an UltraScale+ device and routes the counter using RWRoute.
+     */
+    public static void main(String[] args) {
+        // Extract program options
+        OptionParser p = createOptionParser();
+        OptionSet opts = p.parse(args);
+        boolean verbose = (boolean) opts.valueOf(VERBOSE_OPT);
+        if (opts.has(HELP_OPT)) {
+            printHelp(p);
+            return;
+        }
+
+        CodePerfTracker t = verbose ? new CodePerfTracker(AddSubGenerator.class.getSimpleName(),true).start("Init") : null;
+
+        String partName = (String) opts.valueOf(PART_OPT);
+        String designName = (String) opts.valueOf(DESIGN_NAME_OPT);
+        String outputDCPFileName = (String) opts.valueOf(OUT_DCP_OPT);
+        double clkPeriodConstraint = (double) opts.valueOf(CLK_CONSTRAINT_OPT);
+        int width = (int) opts.valueOf(WIDTH_OPT);
+        String sliceName = (String) opts.valueOf(SLICE_SITES_OPT);
+        boolean countDown = (boolean) opts.valueOf(COUNT_DOWN_OPT);
+        long step = (long) opts.valueOf(STEP_OPT);
+        long initValue = (long) opts.valueOf(INIT_OPT);
```

Comment:
It looks like you have some error checking for the part below, but let's add some more.  This could also be added to the test as well.  Things like a valid/supported width (greater than 1, less than some supported value).  If the `initValue` is larger than what the counter width can represent, etc.  The slice name can also be checked once the device has been loaded.